src/main/scala/cromwell/engine/WorkflowDescriptor.scala

```diff
@@ -82,79 +91,116 @@ case class WorkflowDescriptor(id: WorkflowId,
   lazy val writeToCache = configCallCaching && optionCacheWriting
   lazy val readFromCache = configCallCaching && optionCacheReading
 
-  lazy val workflowLogger = props.get("LOG_MODE") match {
-    case Some(x) if x.toUpperCase.contains("SERVER") => makeFileLogger(
-      Paths.get(props.getOrElse("LOG_ROOT", ".")),
-      s"workflow.$id.log",
-      Level.toLevel(props.getOrElse("LOG_LEVEL", "debug"))
-    )
-    case _ => NOPLogger.NOP_LOGGER
+  lazy val workflowLogName = s"workflow.$id.log"
+  lazy val workflowLogPath = props.get("LOG_MODE") match {
+    case Some(x) if x.toUpperCase.contains("SERVER") =>
+      Option(File(props.getOrElse("LOG_ROOT", ".")) / workflowLogName) map (_.path)
+    case _ => None
+  }
+  lazy val workflowLogger = workflowLogPath match {
+    case Some(path) => makeFileLogger(path, workflowLogName, Level.toLevel(props.getOrElse("LOG_LEVEL", "debug")))
+    case None => NOPLogger.NOP_LOGGER
   }
 
   lazy val workflowRootPath = wfContext.root.toPath(gcsFilesystem)
-  def workflowRootPathWithBaseRoot(rootPath: String): Path =
+  def workflowRootPathWithBaseRoot(rootPath: String): Path = {
     WorkflowDescriptor.buildWorkflowRootPath(rootPath, name, id).toPath(gcsFilesystem)
+  }
 
-  def copyWorkflowOutputs(implicit executionContext: ExecutionContext): Future[Unit] = {
+  def copyWorkflowOutputs(workflowMetadataResponse: WorkflowMetadataResponse)
```

Comment:
Heh, I backed off a number of times from various rabbit holes. You found another one here. :)

Right now, a lot of copying logic is almost magic:rainbow:, using various implicits to inject custom GCS copy handlers into `java.nio.Path` objects. The singleton-ish `Backend` is instead used to validate per-WorkflowDescriptor options and runtime attributes, but does NOT do any actually copying, especially across filesystems.

I expect our new backend repo will create massive changes, and I only _assume_ someone else is already performing surgery to separate backend usage out of the workflow descriptor. But I can continue down that path of moving "file copying" into a/the backend in this PR, if requested.